packages/core/src/global_context.rs

```diff
@@ -310,34 +310,38 @@ pub fn use_hook<State: Clone + 'static>(initializer: impl FnOnce() -> State) ->
     Runtime::with_current_scope(|cx| cx.use_hook(initializer)).unwrap()
 }
 
-/// Get the current render since the inception of this component
+/// Get the current render since the inception of this component.
 ///
-/// This can be used as a helpful diagnostic when debugging hooks/renders, etc
+/// This can be used as a helpful diagnostic when debugging hooks/renders, etc.
 pub fn generation() -> usize {
     Runtime::with_current_scope(|cx| cx.generation()).unwrap()
 }
 
-/// Get the parent of the current scope if it exists
+/// Get the parent of the current scope if it exists.
 pub fn parent_scope() -> Option<ScopeId> {
     Runtime::with_current_scope(|cx| cx.parent_id())
         .ok()
         .flatten()
 }
 
-/// Mark the current scope as dirty, causing it to re-render
+/// Mark the current scope as dirty, causing it to re-render.
 pub fn needs_update() {
     let _ = Runtime::with_current_scope(|cx| cx.needs_update());
 }
 
-/// Mark the current scope as dirty, causing it to re-render
+/// Mark the current scope as dirty, causing it to re-render.
 pub fn needs_update_any(id: ScopeId) {
     let _ = Runtime::with_current_scope(|cx| cx.needs_update_any(id));
 }
 
-/// Schedule an update for the current component
+/// Schedule an update for the current component.
 ///
 /// Note: Unlike [`needs_update`], the function returned by this method will work outside of the dioxus runtime.
 ///
+/// Note: The function returned by this method will schedule an update for the current component even if it has already updated between when `schedule_update` was called and when the returned function is called.
+/// If the desired behavior is to invalidate the current rendering of the current component (and on-op if already invalidated)
```

Comment:
Thanks this is more clear. I think on-op should be no-op

Reply:
Fixed.